dynamical_scaling.c

```diff
@@ -0,0 +1,101 @@
+#include<stdio.h>
+#include<stdlib.h>
+typedef struct vector
+{
+    int *arr;
+    int curLen;
+    int totalSize;
+} vector;
+
+vector* createVector(int totalSize)
+{
+    vector* vec = (vector *)malloc(sizeof(vector));
+    vec->arr = (int *)malloc(sizeof(int)*totalSize);
+    vec->curLen = 0;
+    vec->totalSize = totalSize;
+    return vec;
+}
+
+void resize(vector *vec)
+{
+    vec->arr=(int*)realloc(vec->arr,sizeof(int)*(vec->curLen+1));
+    return;
+}
+void insert(vector *vec,int key)
+{
+
+    resize(vec);
+    vec->arr[vec->curLen]=key;
+    vec->curLen=vec->curLen+1;
+    return;
+}
+
+void display(vector *vec)
+{
+    int index;
+    for(index=0; index<vec->curLen; index++)
+    {
+        printf("%d\t",vec->arr[index]);

```

Comment:
Display should get VectorHead
It should use vectorHead->printType to print vectorHead->vec->arr[index]